### ISSUE TITLE
Jetpack Connection: list of connection services should be a list in the markup

### ISSUE THREAD

**Issue by monsieur-z** [Automattic/jetpack#35486]
In the "Connect Account" page, the list of services should be an actual list in the markup.

![Image](https://github.com/Automattic/jetpack/assets/1620183/5169bfcf-a0b1-45d9-8007-6ced93c8177e)

Issue reported in p1HpG7-qVW-p2.
